CONTRIBUTING.md

```diff
@@ -2,15 +2,184 @@ Guidelines for Contributing
+    -   single module or packages with `make` `-f` `posix.mak`
+        `std/algorithm/comparison.test` or `make` `-f` `posix.mak`
+        `std/algorithm.test`
+    -   all tests with `make` `-f` `posix.mak` `unittest` (add `-j`
+        `NUMBER_OF_CORES` if you have multiple cores)
+    -   for small changes in a single module faster with `rdmd` `-main`
+        `-unittest` `comparison.d` (be careful, this links to your
+        current Phobos version)
+    -   for Windows have a look at [building Phobos under
+        windows](https://wiki.dlang.org/Starting_as_a_Contributor#Windows_2))
+-   Do your tests cover all cases? (you can check code coverage in the
+    resulting `module.lst` of `rdmd` `-cov` `-main` `-unittest`
+    `module.d`)
+
+Review process
+--------------
+
+-   Every PR *must* pass all tests for all supported platforms of
+    [autotester](https://wiki.dlang.org/Git_Commit_Tester) before being merged
+-   Big additions (new modules) should go to `std.experimental` first
+    (after they pass the [review process](https://wiki.dlang.org/Review/Process))
+-   Smaller additions like individual functions can be merged directly
+    after [@andralex](https://github.com/andralex) approves
+-   Refactoring or bug fixes just need approval (LGTM) from two
+    reviewers and enough time passed (usually 2-3 days, depending on
+    weekends and holidays) to give everyone an opportunity to shout
+-   For low-level changes the other compiler devs (GDC and LLVM) should
+    be consulted
+-   Trivial changes (e.g. small documentation enhancements) should be marked with
+	`[Trivial]` as title prefix
+-   If your PR is stalled (no activity within a couple of days) and you addressed
+    all concerns, don't hesitate to ping your reviewers
+-   See [rebasing](https://wiki.dlang.org/Starting_as_a_Contributor#Rebasing) if you
+    need to sync your fork with master
+
+Naming guidelines
+-----------------
+
+### General advices
+
+-   Phobos uses a distinct [naming scheme](https://wiki.dlang.org/Naming_conventions)
+-   Use a Capitalized short (50 chars or less) Git commit summary
+-   If necessary, wrap your Git explanation paragraph [to 72 chars](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html)
+-   Do you use clear variable names?
+-   Does your PR have a concise and informative title?
+-   Are your Git messages clear and informative (otherwise use `git commit --amend`
+	for a single commit or `git rebase` for multiple commits)?
+-   Avoid having many commits per PR (commits should reflect actual changes, not design iterations) - see
+    [squashing](https://wiki.dlang.org/Starting_as_a_Contributor#Someone_asked_me_to_squash_my_commits.2C_what_does_that_mean.3F) to combine commits
+
+### Issues :fire:
+
+-   Bug fix PRs should have a separate commit for each bugzilla issue
+    that it fixes
+-   The message of the commit actually fixes the bug should start with
+    "Fix issue X" where X is the number of the corresponding
```

Comment:
Actually most people use "Fix issue X - [issue title from bugzilla]". I also prefer it because it's more informative.